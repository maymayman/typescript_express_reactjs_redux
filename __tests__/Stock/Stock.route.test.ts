import * as supertest from 'supertest';
import app from '../../src';

const request = supertest(app);

jest.mock('../../src/models/', () => ({
    default: {
      Stocks: class {
        public save = jest.fn().mockResolvedValueOnce({
            stock_code:"132a3",
            stock_name:"viettel",
            stock_price:100.11
        });
        constructor() {
        } 
      }
    }
  })
);

afterAll(() => {
    jest.resetAllMocks();
});

describe('POST STOCK',()=>{
    it('POST /stocks - create stock validate stock_code',async ()=>{
        const stock ={
            stock_name:"viettel",
            stock_price:100.11
        }
        const result = await request.post('/stocks').send(stock);

        expect(result.status).toEqual(400);
        if (result.error) {
            expect(result.error.text).toMatchSnapshot();
        }
    })
    it('POST /stocks - create stock validate success',async ()=>{
        const stock ={
            stock_code:"132a3",
            stock_name:"viettel",
            stock_price:100.11
        }
        const result = await request.post('/stocks').send(stock);
        expect(result.status).toEqual(200);
        expect(result.body).toEqual({
            stock_code:"132a3",
            stock_name:"viettel",
            stock_price:100.11
        })
    })
})