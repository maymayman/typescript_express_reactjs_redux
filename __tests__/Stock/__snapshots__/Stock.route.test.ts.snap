// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST STOCK POST /stocks - create stock validate stock_code 1`] = `"{\\"message\\":\\"\\\\\\"stock_code\\\\\\" is required\\"}"`;
